libs/entitlement/src/main/java/org/elasticsearch/entitlement/runtime/policy/PolicyManager.java

```diff
@@ -200,24 +200,25 @@ public void checkNetworkAccess(Class<?> callerClass, int actions) {
             return;
         }
 
-        ModuleEntitlements entitlements = getEntitlements(requestingClass);
+        var actionsString = NetworkEntitlement.printActions(actions);
```

Comment:
This is on a critical path happening every time we check network entitlements, maybe entitlementInfo should be passed lazy as supplier or an arbitrary class with a toString()?